src/codegen/compile_enhanced.ml

```diff
@@ -4212,7 +4231,12 @@ module Text = struct
     get_blob ^^ Blob.as_ptr_len env ^^
     E.call_import env "rts" "utf8_valid" ^^
     Bool.from_rts_int32 ^^
-    E.if1 I64Type (Opt.inject_simple env get_blob) (Opt.null_lit env)
+    E.if1 I64Type
+      (get_blob ^^ Blob.as_ptr_len env ^^
+       of_ptr_size env ^^ (* creates text blob *)
+       set_blob ^^
```

Comment:
(Maybe we could just use Blob.copy env Tagged.B Tagged.T (or similar) here... (and it compile_classical).)

Reply:
Thank you for refactoring suggestion!